TM1py/Services/ElementService.py

```diff
@@ -142,7 +141,27 @@ def get_elements_by_level(self, dimension_name: str, hierarchy_name: str, level:
             "/api/v1/Dimensions('{}')/Hierarchies('{}')/Elements?$select=Name&$filter=Level eq {}",
             dimension_name,
             hierarchy_name,
-            level)
+            str(level))
+        response = self._rest.GET(url, **kwargs)
+        return [e["Name"] for e in response.json()['value']]
+
+    def get_elements_by_wildcard(self, dimension_name: str, hierarchy_name: str,
+                                 wildcard: str, level: int = None, **kwargs) -> List[str]:
+        """ Get all element names filtered by wildcard (CaseInsensitive) and level in a hierarchy
+
+        :param dimension_name: Name of the dimension
+        :param hierarchy_name: Name of the hierarchy
+        :param wildcard: wildcard to filter
+        :param level: Level to filter
+        :return: List of element names
+        """
+        filter_elements = format_url("contains(tolower(Name),tolower('{}'))", wildcard)
+        if isinstance(level, int):
```

Comment:
what if someones passes `level` as str or float? Shall we raise a `ValueError` indicating that we expect an integer?

Reply:
Actually `int` or `str` or `float` should work.. I will update the condition to:
```python
if level is not None:
```